### ISSUE TITLE
Wallet - ERC-20 (and perhaps NFT) transactions no longer appearing in transaction history

### ISSUE THREAD

**Issue by John-44** [status-im/status-react#12802]
To reproduce:
- Send some ETH to an account
- Sent some SNT to the same account

Expected behaviour:
- User should be able to see all three transactions in their transaction history

Currently incorrect behaviour (using iOS release 1.17):
- The user only sees the ETH transaction under History, the SNT transaction isn't shown

Desired resolution:
- History should display all transactions in and out of the account, including ERC-20s and NFTs, not just ETH

**Comment by rasom** [comment]
Just tried to check history using some watch only account, works as expected. 
@John-44 Could you please share an account for which history is not loaded? Also if you create a new multiaccount and add that one as a watch only account, does history appear in this case?

**Comment by John-44** [comment]
@rasom it was a standard account (not a watch only account).  I've checked again, and the SNT transactions have now appeared.  So the issue seems to be that ETH transactions appear in the transaction history immediately but SNT transactions seem to take a while to appear in the transaction history.

**Comment by churik** [comment]
Also rechecked using ERC-20 (AST token) - transaction appeared after sending (can't check SNT due to insane tx fee).
Also rechecked on STT with a low tx fee, still can see ERC-20 (even after re-login)

**Comment by rasom** [comment]
@John-44 correct me if I'm wrong, but transactions were sent to that account from somewhere else, right? I mean, not for Status app

**Comment by John-44** [comment]
@rasom the delayed SNT transactions that didn't show up immediately were both an incoming and an outgoing transaction.  The transactions in question showed in etherscan immediately

**Comment by rasom** [comment]
@John-44 
So it depends whether that transaction was shown on etherscan as a pending transaction or it was confirmed already. 
An outgoing transaction which is sent form the Status app will appear in the history as a pending transaction right after sending, but outgoing/incoming txs which are sent from somewhere else will appear (in the worst case) in 20 mins after being included (we have 20mins interval for checking recent history). Currently there are no simple means of watching pending txs via ethereum's RPC. 

In other words, we have an issue only in case if we can't find an outgoing tx (which is sent from Status) right after sending it. Otherwise it is likely due to "scanning" interval.

**Comment by John-44** [comment]
Next time I make an SNT transaction I’ll get a timer out and watch closely and time everything and report back. Thanks for looking into this

**Comment by churik** [comment]
Cross-reference with https://github.com/status-im/status-mobile/issues/12256 
Don't think it makes sense to keep it open in terms of re-design app and moving to Pokt
Please re-open if still makes sense from your POV.

The label `recheck` is added in order to keep an eye on it while testing tx history on the new design.
